code/datums/actions/xeno_action.dm

```diff
@@ -10,6 +10,10 @@
 	var/keybind_signal
 	var/cooldown_id
 
+	//Boiler
+	var/corrosive_ammo = 0
```

Comment:
Oh. You have multiple actions dealing with the same ability. This is bad and requires a component.
Guess I'll have to make one for you.

Reply:
I'm afraid I don't follow the multiple actions part. There is only one ability that makes the ammo.

Reply:
And there is another different ability that consumes it.

Reply:
Which is why I felt it was important to have it stored on the mob, cause it makes sense to keep things there. A pouch that stores the globs if you will.

Reply:
Yea, usually that'd be right. But the idea is to make actions generic, so that we can apply it to anything, other xeno castes or whatever.
That's why we'll be doing a component which stores this information.
The logic is that if we slap the component to a mob it adds these actions and manages them.